### PR TITLE
#2522 - Map PD/PPD Status to COE screen

### PR DESCRIPTION
**Pull request by andrepestana-aot**
- Added student's disability statuses to COE:
  - image
  - Applied the statuses as per @HRAGANBC's [comment](https://app.zenhub.com/workspaces/student-information-management-system-5fce9df5aa1b45000e937014/issues/gh/bcgov/sims/2522#issuecomment-1900918994);
  - Fixed e2e tests;
- Changed E-Cert to send PD/PPD application flag;
- Removed `pd_verified` from `sims.students` table;
- Mapped PD/PPD statuses from SFAS to the table `sims.sfas_individuals`;
  - Found an issue with the `ssh2-sftp-client`: after updating the lib, the [`list()` method requires now a filter function instead of a regex](https://github.com/theophilusx/ssh2-sftp-client/releases/tag/v9.0.0):
    - [Updated the code](https://github.com/bcgov/SIMS/pull/2691/files#diff-d2bb64aaf062ab1fc3d7eecaa44f0dc71aa67895d7f2e98b689130363ecf0ec8R79) to fix that; 
    - Updated `@types/ssh2-sftp-client` version and;
    - Created an [e2e test ticket](https://app.zenhub.com/workspaces/student-information-management-system-5fce9df5aa1b45000e937014/issues/gh/bcgov/sims/2717) for this issue;
  - Changed the deprecated 
 method `substr()` calls to `substring()` in [sfas-individual-record.ts](https://github.com/bcgov/SIMS/pull/2691/files#diff-4fce98afba8d44e99de50a807e110f198e156934301a7678d01a39e70d0aeeb6).

### CODE REVIEW

File: sources/packages/backend/apps/api/src/route-controllers/confirmation-of-enrollment/_tests_/e2e/confirmation-of-enrollment.institutions.getApplicationForCOE.e2e-spec.ts [hunk 1]
```diff
@@ -165,7 +173,7 @@ describe("ConfirmationOfEnrollmentInstitutionsController(e2e)-getApplicationForC
       disbursementSchedule: firstDisbursementSchedule,
       studentAssessment: application.currentAssessment,
     });
-    await disbursementOverawardRepo.save(overaward);
+    await db.disbursementOveraward.save(overaward);
```
Comment (sh16011993) [inline sources/packages/backend/apps/api/src/route-controllers/confirmation-of-enrollment/_tests_/e2e/confirmation-of-enrollment.institutions.getApplicationForCOE.e2e-spec.ts:176]: 👍

File: sources/packages/backend/apps/api/src/services/disbursement-schedule/disbursement-schedule-service.ts [hunk 2]
```diff
@@ -138,10 +138,12 @@ export class DisbursementScheduleService extends RecordDataModelService<Disburse
         "application.applicationStatus",
         "application.id",
         "application.pirStatus",
+        "application.data",
```
Comment (sh16011993) [inline sources/packages/backend/apps/api/src/services/disbursement-schedule/disbursement-schedule-service.ts:141]: Do we need to return the entire data?

File: sources/packages/backend/apps/db-migrations/src/sql/SFASIndividuals/Add-ppd-status-ppd-status-date.sql [hunk 3]
```diff
@@ -0,0 +1,13 @@
+ALTER TABLE
+    sims.sfas_individuals
+ADD
+    COLUMN ppd_status BOOLEAN;
+
+COMMENT ON COLUMN sims.sfas_individuals.ppd_status IS 'Persistent or Prolonged Disability Flag (individual.ppd_flg).';
```
Comment (sh16011993) [inline sources/packages/backend/apps/db-migrations/src/sql/SFASIndividuals/Add-ppd-status-ppd-status-date.sql:6]: Very minor: `ppd_flag`

File: sources/packages/web/src/services/http/dto/ConfirmationOfEnrolment.dto.ts [hunk 4]
```diff
@@ -53,6 +54,8 @@ export interface ApplicationDetailsForCOEAPIOutDTO {
   coeStatusClass: StatusChipTypes;
   maxTuitionRemittanceAllowed: number;
   hasOverawardBalance: boolean;
+  disabilityProfileStatus: DisabilityStatus;
+  disabilityApplicationStatus: DisabilityStatus;
```
Comment (sh16011993) [inline sources/packages/web/src/services/http/dto/ConfirmationOfEnrolment.dto.ts:58]: Are both the types in line 57 and 58 supposed to be the same?

**Review by sh16011993 — COMMENTED** [review]
Nice Work @andrepestana-aot  👍

File: sources/packages/backend/apps/api/src/services/disbursement-schedule/disbursement-schedule-service.ts (hunk 2 above)
Comment (andrepestana-aot) [reply sources/packages/backend/apps/api/src/services/disbursement-schedule/disbursement-schedule-service.ts:141]: `application.data` is a jsonb. I did some research about it and I don't think it's possible to retrieve only that property of the jsonb through TypeORM as far as I could learn about it.

File: sources/packages/web/src/services/http/dto/ConfirmationOfEnrolment.dto.ts (hunk 4 above)
Comment (andrepestana-aot) [reply sources/packages/web/src/services/http/dto/ConfirmationOfEnrolment.dto.ts]: Good catch. Both should be strings.

File: sources/packages/backend/apps/api/src/route-controllers/confirmation-of-enrollment/models/confirmation-of-enrollment.dto.ts [hunk 5]
```diff
@@ -33,6 +33,8 @@ export class ApplicationDetailsForCOEAPIOutDTO {
   applicationProgramDelivery: string;
   maxTuitionRemittanceAllowed: number;
   hasOverawardBalance: boolean;
+  disabilityProfileStatus: string;
```
Comment (dheepak-aot) [inline sources/packages/backend/apps/api/src/route-controllers/confirmation-of-enrollment/models/confirmation-of-enrollment.dto.ts:36]: Shouldn't his be an Enum?

File: sources/packages/backend/libs/sims-db/src/entities/application.model.ts [hunk 6]
```diff
@@ -426,6 +427,10 @@ export interface ApplicationData {
    * Course Details is present only for Part Time course intensity.
    */
   courseDetails?: CourseDetails[];
+  /**
+   * Application PD/PPD Status.
+   */
+  applicationPDPPDStatus?: string;
 }
```
Comment (lewischen-aot) [inline sources/packages/backend/libs/sims-db/src/entities/application.model.ts:434]: I think we can use enum `ApplicationDisabilityStatus` instead of string here. Please let me know your thoughts.

File: sources/packages/backend/apps/api/src/services/disbursement-schedule/disbursement-schedule-service.ts (hunk 2 above)
Comment (dheepak-aot) [reply sources/packages/backend/apps/api/src/services/disbursement-schedule/disbursement-schedule-service.ts:141]: @andrepestana-aot there is a way to get a particular property of jsonb using query builder in type orm. 

image

But if we pick a property, then we should use `getRawOne` instead of getOne. 

Also this option cannot be used in typeorm find* query.

File: sources/packages/backend/apps/db-migrations/src/sql/SFASIndividuals/Add-ppd-status-ppd-status-date.sql [hunk 7]
```diff
@@ -0,0 +1,13 @@
+ALTER TABLE
+    sims.sfas_individuals
+ADD
+    COLUMN ppd_status BOOLEAN;
```
Comment (dheepak-aot) [inline sources/packages/backend/apps/db-migrations/src/sql/SFASIndividuals/Add-ppd-status-ppd-status-date.sql:4]: @andrewsignori-aot @guru-aot @andrepestana-aot  I have a confusion here, should we add a new column or  should we change the existing coulmn type to enum on sfas_individuals.?

File: sources/packages/backend/apps/db-migrations/src/sql/SFASIndividuals/Add-ppd-status-ppd-status-date.sql [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+
+COMMENT ON COLUMN sims.sfas_individuals.ppd_status IS 'Persistent or Prolonged Disability Flag (individual.ppd_flg).';
+
+ALTER TABLE
+    sims.sfas_individuals
+ADD
+    COLUMN ppd_status_date DATE;
```
Comment (dheepak-aot) [inline sources/packages/backend/apps/db-migrations/src/sql/SFASIndividuals/Add-ppd-status-ppd-status-date.sql:11]: Is this a ppd effective date? if so is it appropriate to map it to sims effective date??

File: sources/packages/backend/apps/db-migrations/src/migrations/1705441135641-RemovePdVerifiedColumn.ts [hunk 9]
```diff
@@ -0,0 +1,15 @@
+import { MigrationInterface, QueryRunner } from "typeorm";
+import { getSQLFileData } from "../utilities/sqlLoader";
+
+export class RemovePdVerifiedColumn1705441135641 implements MigrationInterface {
```
Comment (dheepak-aot) [inline sources/packages/backend/apps/db-migrations/src/migrations/1705441135641-RemovePdVerifiedColumn.ts:4]: PDVerified? (Same in file name)

File: sources/packages/backend/libs/integrations/src/esdc-integration/e-cert-integration/e-cert-file-handler.ts [hunk 10]
```diff
@@ -264,12 +264,10 @@ export abstract class ECertFileHandler extends ESDCFileHandler {
       postalCode: addressInfo.postalCode,
       email: student.user.email,
       gender: student.gender,
-      // TODO: getting the information directly from the student profile since there is no
-      // direction from the business at this moment from where it should actually come.
-      ppdFlag: student.studentPDVerified,
       maritalStatus: application.relationshipStatus,
       studentNumber: application.studentNumber,
       awards,
+      ppdFlag: application.data.applicationPDPPDStatus,
```
Comment (dheepak-aot) [inline sources/packages/backend/libs/integrations/src/esdc-integration/e-cert-integration/e-cert-file-handler.ts:270]: applicationPDPPDStatus?

File: sources/packages/backend/libs/integrations/src/sfas-integration/sfas-files/sfas-individual-record.ts [hunk 11]
```diff
@@ -12,27 +12,27 @@ export class SFASIndividualRecord extends SFASRecordIdentification {
    * SFAS Unique id.
    */
   get id(): number {
-    return +this.line.substr(3, 10);
+    return +this.line.substring(3, 13);
   }
```
Comment (lewischen-aot) [inline sources/packages/backend/libs/integrations/src/sfas-integration/sfas-files/sfas-individual-record.ts:16]: I'm thinking maybe we can have a e2e (unit) test to verify that the an SFAS record is correctly saved to an instance since there are quite a few changes here. Please let me know your thoughts.

File: sources/packages/backend/libs/integrations/src/esdc-integration/e-cert-integration/e-cert-file-handler.ts (hunk 10 above)
Comment (andrepestana-aot) [reply sources/packages/backend/libs/integrations/src/esdc-integration/e-cert-integration/e-cert-file-handler.ts:270]: @dheepak-aot I didn't understand this comment. Any typo?

File: sources/packages/backend/apps/api/src/services/disbursement-schedule/disbursement-schedule-service.ts (hunk 2 above)
Comment (andrepestana-aot) [reply sources/packages/backend/apps/api/src/services/disbursement-schedule/disbursement-schedule-service.ts:141]: Yes, @dheepak-aot I know that is possible with the `getRawOne`. I just didn't mention that because it doesn't apply here.

File: sources/packages/backend/libs/integrations/src/esdc-integration/e-cert-integration/e-cert-file-handler.ts (hunk 10 above)
Comment (dheepak-aot) [reply sources/packages/backend/libs/integrations/src/esdc-integration/e-cert-integration/e-cert-file-handler.ts:270]: I meant the property name 
image

File: sources/packages/backend/apps/db-migrations/src/sql/SFASIndividuals/Add-ppd-status-ppd-status-date.sql (hunk 7 above)
Comment (andrepestana-aot) [reply sources/packages/backend/apps/db-migrations/src/sql/SFASIndividuals/Add-ppd-status-ppd-status-date.sql:4]: As far as I understood `sims.sfas_individuals` just keeps the raw data that comes from the integration.

File: sources/packages/backend/apps/db-migrations/src/sql/SFASIndividuals/Add-ppd-status-ppd-status-date.sql (hunk 8 above)
Comment (andrepestana-aot) [reply sources/packages/backend/apps/db-migrations/src/sql/SFASIndividuals/Add-ppd-status-ppd-status-date.sql:11]: As column comment it is the PPD effective date. What do you mean by `sims effective date`? Do you think its name should be `ppd_effective_date`?

File: sources/packages/backend/apps/api/src/route-controllers/confirmation-of-enrollment/models/confirmation-of-enrollment.dto.ts (hunk 5 above)
Comment (andrepestana-aot) [reply sources/packages/backend/apps/api/src/route-controllers/confirmation-of-enrollment/models/confirmation-of-enrollment.dto.ts:36]: I opted for having the enum and the descriptions in the API and just pass it as a string to the view.

File: sources/packages/backend/apps/db-migrations/src/sql/SFASIndividuals/Add-ppd-status-ppd-status-date.sql (hunk 7 above)
Comment (andrewsignori-aot) [reply sources/packages/backend/apps/db-migrations/src/sql/SFASIndividuals/Add-ppd-status-ppd-status-date.sql:4]: I agree with @andrepestana-aot. The sfas* tables can reflect as much as possible the raw data received.

File: sources/packages/backend/libs/sims-db/src/entities/application.model.ts (hunk 6 above)
Comment (lewischen-aot) [reply sources/packages/backend/libs/sims-db/src/entities/application.model.ts]: Thanks for making the change.

File: sources/packages/backend/apps/db-migrations/src/sql/SFASIndividuals/Add-ppd-status-ppd-status-date.sql (hunk 8 above)
Comment (dheepak-aot) [reply sources/packages/backend/apps/db-migrations/src/sql/SFASIndividuals/Add-ppd-status-ppd-status-date.sql:11]: There is a column in students table `disability_status_effective_date` and seeing the story description I thought so.

image

I am ok with column name. My question was is that intended to be mapped to students table to column `disability_status_effective_date`

File: sources/packages/backend/libs/utilities/src/esdc-utils.ts [hunk 12]
```diff
@@ -43,11 +47,11 @@ export function getPartTimeMaritalStatusCode(
 
 /**
  * Get the e-Cert flag for the borrower's persistent or prolonged disability status.
- * @param hasPDorPPD indicates if a borrower has a persistent or prolonged disability.
+ * @param hasPPD indicates if a borrower has a persistent or prolonged disability.
  * @returns "Y" or "N" flag.
  */
-export function getPPDFlag(hasPPD?: boolean): string {
-  return hasPPD === true ? "Y" : "N";
+export function getPPDFlag(hasPPD?: ApplicationDisabilityStatus): string {
```
Comment (dheepak-aot) [inline sources/packages/backend/libs/utilities/src/esdc-utils.ts:53]: Is still the argument optional?

Also hasPPD or even the function name can be renamed right?

File: sources/packages/backend/libs/integrations/src/esdc-integration/e-cert-integration/models/e-cert-integration-model.ts [hunk 13]
```diff
@@ -49,7 +53,7 @@ export interface ECertRecord {
   /**
    * Persistent or prolonged disability flag.
    */
```
Comment (dheepak-aot) [inline sources/packages/backend/libs/integrations/src/esdc-integration/e-cert-integration/models/e-cert-integration-model.ts:55]: Comment can also be adjusted accordingly

File: sources/packages/backend/libs/integrations/src/services/sfas/sfas-individual.service.ts [hunk 14]
```diff
@@ -68,6 +68,10 @@ export class SFASIndividualService
     individual.birthDate = getISODateOnlyString(sfasIndividual.birthDate);
     individual.sin = sfasIndividual.sin;
     individual.pdStatus = sfasIndividual.pdStatus;
+    individual.ppdStatus = sfasIndividual.ppdStatus;
```
Comment (dheepak-aot) [inline sources/packages/backend/libs/integrations/src/services/sfas/sfas-individual.service.ts:71]: To be fair, we need an adjustment during the student creation to consider ppd status from SFAS and that was not included in the ticket. 

We need to create a seperate ticket for the same. 

image

File: sources/packages/backend/libs/sims-db/src/entities/application.model.ts [hunk 15]
```diff
@@ -359,6 +360,7 @@ export class Application extends RecordDataModel {
     },
   )
   applicationOfferingChangeRequest?: ApplicationOfferingChangeRequest[];
+  application: { applicationPDPPDStatus: string };
```
Comment (dheepak-aot) [inline sources/packages/backend/libs/sims-db/src/entities/application.model.ts:363]: Why was this property added? - asking for my understanding

File: sources/packages/backend/libs/sims-db/src/entities/sfas-individual.model.ts [hunk 16]
```diff
@@ -53,6 +53,23 @@ export class SFASIndividual extends BaseModel {
...
+  ppdStatus?: boolean;
+  /**
+   * The date when a PPD status is effective (individual.ppd_status_dte).
+   */
+  @Column({
+    name: "ppd_status_date",
+    type: "date",
+    nullable: true,
+  })
+  ppdStatusDate: string;
```
Comment (dheepak-aot) [inline sources/packages/backend/libs/sims-db/src/entities/sfas-individual.model.ts:72]: Is this also not an optional column?

File: sources/packages/backend/libs/sims-db/src/entities/student-disability-status.type.ts [hunk 17]
```diff
@@ -10,3 +10,14 @@ export enum DisabilityStatus {
   PPD = "PPD",
   Declined = "Declined",
 }
+
+/**
+ *  Disability status description.
+ */
+export const DisabilityStatusDescription: Record<DisabilityStatus, string> = {
```
Comment (dheepak-aot) [inline sources/packages/backend/libs/sims-db/src/entities/student-disability-status.type.ts:17]: Why is this conversion required in API?

For UI purpose we already have a composable 
image

File: sources/packages/backend/libs/sims-db/src/entities/application-disability-status.type.ts [hunk 18]
```diff
@@ -0,0 +1,23 @@
...
+export enum ApplicationDisabilityStatus {
+  yes = "yes",
+  noIWantToAccessOtherFundingTypes = "noIWantToAccessOtherFundingTypes",
+  noIDoNotHaveADisability = "noIDoNotHaveADisability",
+}
+
+/**
+ * Application disability status description.
+ */
+export const ApplicationDisabilityStatusDescription: Record<
```
Comment (dheepak-aot) [inline sources/packages/backend/libs/sims-db/src/entities/application-disability-status.type.ts:13]: IMO the conversion is not required from API perspective. We can do this at Vue level.

File: sources/packages/forms/src/form-definitions/confirmsstudentenrollment.json [hunk 19]
```diff
@@ -412,6 +412,595 @@
...
+              "tag": "h2",
+              "className": "category-header-large primary-color",
+              "attrs": [
+                {
+                  "attr": "",
+                  "value": ""
+                }
+              ],
+              "content": "Disability Information",
+              "refreshOnChange": false,
```
Comment (dheepak-aot) [inline sources/packages/forms/src/form-definitions/confirmsstudentenrollment.json:428]: We use to enable `refreshOnChange` to true on HTML components using form data values.

With the current editor, we don't have the option to change it from UI. But I believe we must enable from definition. 

image

File: sources/packages/backend/libs/integrations/src/services/ssh/sftp-integration-base.ts [hunk 20]
```diff
@@ -74,7 +74,10 @@ export abstract class SFTPIntegrationBase<DownloadType> {
     let filesToProcess: Client.FileInfo[];
     const client = await this.getClient();
     try {
-      filesToProcess = await client.list(remoteDownloadFolder, fileRegexSearch);
+      filesToProcess = await client.list(
```
Comment (dheepak-aot) [inline sources/packages/backend/libs/integrations/src/services/ssh/sftp-integration-base.ts:77]: Thanks for taking care of this.

File: sources/packages/forms/src/form-definitions/confirmsstudentenrollment.json [hunk 21]
```diff
@@ -412,6 +412,595 @@
           "input": false,
           "tableView": false,
           "components": [
+            {
+              "label": "HTML",
```
Comment (dheepak-aot) [inline sources/packages/forms/src/form-definitions/confirmsstudentenrollment.json:416]: Outside the PR context. During the grooming, we decided to keep converting the COE page from form.io to Vuetify outside the parttime release scope but still have a ticket created for the same to not lose track of it.

**Comment by dheepak-aot** [comment]
Great job @andrepestana-aot. Thanks for additional efforts to fix the sftp client filter. Please have a look at the comments.

File: sources/packages/backend/libs/utilities/src/esdc-utils.ts (hunk 12 above)
Comment (sh16011993) [reply sources/packages/backend/libs/utilities/src/esdc-utils.ts:53]: Seems to me that the function name is right since it is not returning a Boolean.

File: sources/packages/backend/libs/sims-db/src/entities/application-disability-status.type.ts (hunk 18 above)
Comment (andrepestana-aot) [reply sources/packages/backend/libs/sims-db/src/entities/application-disability-status.type.ts:13]: IMO it should be in only one place and with the API sending that to the view we can avoid duplicating this logic in the view.

File: sources/packages/backend/apps/db-migrations/src/sql/SFASIndividuals/Add-ppd-status-ppd-status-date.sql (hunk 7 above)
Comment (guru-aot) [reply sources/packages/backend/apps/db-migrations/src/sql/SFASIndividuals/Add-ppd-status-ppd-status-date.sql:4]: Yes i believe that we can have the raw data what is received.

File: sources/packages/backend/libs/sims-db/src/entities/student-disability-status.type.ts (hunk 17 above)
Comment (andrepestana-aot) [reply sources/packages/backend/libs/sims-db/src/entities/student-disability-status.type.ts:17]: Yes, ok. I'm going to change to use the UI logic.

File: sources/packages/backend/libs/sims-db/src/entities/application.model.ts (hunk 15 above)
Comment (andrepestana-aot) [reply sources/packages/backend/libs/sims-db/src/entities/application.model.ts]: I think it was added by accident :) removing.

File: sources/packages/backend/libs/integrations/src/services/ssh/sftp-integration-base.ts [hunk 22]
```diff
@@ -74,7 +74,10 @@ export abstract class SFTPIntegrationBase<DownloadType> {
...
+        remoteDownloadFolder,
+        (item: Client.FileInfo) => fileRegexSearch.test(item.name),
```
Comment (guru-aot) [inline sources/packages/backend/libs/integrations/src/services/ssh/sftp-integration-base.ts:79]: 👍

File: sources/packages/backend/apps/api/src/route-controllers/confirmation-of-enrollment/models/confirmation-of-enrollment.dto.ts (hunk 5 above)
Comment (andrepestana-aot) [reply sources/packages/backend/apps/api/src/route-controllers/confirmation-of-enrollment/models/confirmation-of-enrollment.dto.ts]: It was changed to use the enum.

File: sources/packages/backend/libs/utilities/src/esdc-utils.ts (hunk 12 above)
Comment (andrepestana-aot) [reply sources/packages/backend/libs/utilities/src/esdc-utils.ts:53]: I changed the argument name to `applicationDisabilityStatus`.

File: sources/packages/backend/libs/integrations/src/services/sfas/sfas-individual.service.ts (hunk 14 above)
Comment (andrepestana-aot) [reply sources/packages/backend/libs/integrations/src/services/sfas/sfas-individual.service.ts:71]: I did that adjust in this ticket.

File: sources/packages/backend/apps/api/src/services/student/student.service.ts [hunk 23]
```diff
@@ -680,16 +683,43 @@ export class StudentService extends RecordDataModelService<Student> {
...
+      return DisabilityStatus.PPD;
     }
     return DisabilityStatus.NotRequested;
   }
 
+  /**
+   * Gets the disability effective date based on the disability status.
+   * If the disability status is a pd status, the disability effective date
+   * should be the current date.
+   * If the disability status is a ppd status, the disability effective date should the
```
Comment (andrewsignori-aot) [inline sources/packages/backend/apps/api/src/services/student/student.service.ts:706]: Missing be in the sentence "effective date should BE the".

File: sources/packages/backend/apps/api/src/services/student/student.service.ts [hunk 24]
```diff
@@ -680,16 +683,43 @@ export class StudentService extends RecordDataModelService<Student> {
...
+    if (pdStatus) {
       return DisabilityStatus.PD;
+    } else if (ppdStatus) {
+      return DisabilityStatus.PPD;
     }
     return DisabilityStatus.NotRequested;
   }
 
+  /**
+   * Gets the disability effective date based on the disability status.
```
Comment (andrewsignori-aot) [inline sources/packages/backend/apps/api/src/services/student/student.service.ts:703]: It is minor, but while commenting, acronyms should be upper case like PD/PPD.

File: sources/packages/backend/apps/api/src/services/student/student.service.ts [hunk 25]
```diff
@@ -680,16 +683,43 @@ export class StudentService extends RecordDataModelService<Student> {
...
+   * should be the current date.
+   * If the disability status is a ppd status, the disability effective date should the
+   * ppd status date.
+   * @param ppdStatusDate ppd status date.
+   * @param disabilityStatus student disability status.
+   */
+  getDisabilityEffectiveDate(
+    ppdStatusDate: string,
+    disabilityStatus: DisabilityStatus,
+  ): Date {
```
Comment (andrewsignori-aot) [inline sources/packages/backend/apps/api/src/services/student/student.service.ts:714]: The return type can be changed to `Date | null`.

File: sources/packages/backend/apps/db-migrations/src/migrations/1705441135641-RemovePDVerifiedColumn.ts [hunk 26]
```diff
@@ -0,0 +1,15 @@
+import { MigrationInterface, QueryRunner } from "typeorm";
+import { getSQLFileData } from "../utilities/sqlLoader";
+
+export class RemovePDVerifiedColumn1705441135641 implements MigrationInterface {
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(
+      getSQLFileData("Remove-StudentPDVerified.sql", "Student"),
```
Comment (andrewsignori-aot) [inline sources/packages/backend/apps/db-migrations/src/migrations/1705441135641-RemovePDVerifiedColumn.ts:7]: Not part of this PR but I believe that we should simplify the DB migrations following the below convention, adding the "rollback" as a prefix.
- "Add-something-new-to-db"
- "Rollback-add-something-new-to-db"

Sometimes the rollback file name can be misleading.
@dheepak-aot @andrepestana-aot @guru-aot @sh16011993 @lewischen-aot thoughts?

Comment (andrewsignori-aot) [inline sources/packages/backend/apps/db-migrations/src/migrations/1705441135641-RemovePDVerifiedColumn.ts:7]: We should follow the kebab case for "Remove-StudentPDVerified.sql" and "Add-StudentPDVerified.sql".

Comment (andrepestana-aot) [reply sources/packages/backend/apps/db-migrations/src/migrations/1705441135641-RemovePDVerifiedColumn.ts:7]: Yes, maybe we should be even more generic and use like "change" and "rollback" because sometimes in the db change we do an update or delete.

Comment (andrepestana-aot) [reply sources/packages/backend/apps/db-migrations/src/migrations/1705441135641-RemovePDVerifiedColumn.ts:7]: yes, that file already existed than I didn't want to change its name but I think I can do that with no harm if I change the references. Will change that.

File: sources/packages/backend/apps/db-migrations/src/sql/Student/Remove-StudentPDVerified.sql
Comment (andrewsignori-aot) [inline sources/packages/backend/apps/db-migrations/src/sql/Student/Remove-StudentPDVerified.sql:1]: Ideally, we do not reuse or change past migrations.

File: sources/packages/backend/apps/db-migrations/src/migrations/1705441135641-RemovePDVerifiedColumn.ts (hunk 26 above)
Comment (sh16011993) [reply sources/packages/backend/apps/db-migrations/src/migrations/1705441135641-RemovePDVerifiedColumn.ts]: I agree @andrewsignori-aot. Adding a `rollback` prefix simplifies finding the revert script.

File: sources/packages/backend/libs/integrations/src/esdc-integration/msfaa-integration/msfaa.integration.service.ts [hunk 27]
```diff
@@ -125,7 +125,7 @@ export class MSFAAIntegrationService extends SFTPIntegrationBase<MSFAASFTPRespon
             );
       filesToProcess = await client.list(
         `${this.esdcConfig.ftpResponseFolder}`,
-        pattern,
+        (item: Client.FileInfo) => pattern.test(item.name),
```
Comment (andrewsignori-aot) [inline sources/packages/backend/libs/integrations/src/esdc-integration/msfaa-integration/msfaa.integration.service.ts:128]: Thanks for doing the fix ❤️

File: sources/packages/backend/libs/integrations/src/sfas-integration/sfas-files/sfas-individual-record.ts [hunk 28]
```diff
@@ -78,76 +78,95 @@ export class SFASIndividualRecord extends SFASRecordIdentification {
...
+      case "N":
+        return false;
+      default:
+        return null;
+    }
+  }
+  /**
+   * The date when a PPD status is effective (individual.ppd_status_dte).
+   */
+  get ppdStatusDate(): Date {
```
Comment (andrewsignori-aot) [inline sources/packages/backend/libs/integrations/src/sfas-integration/sfas-files/sfas-individual-record.ts:97]: Please adjust the return type to `Date | null`.

File: sources/packages/backend/apps/db-migrations/src/migrations/1705441135641-RemovePDVerifiedColumn.ts (hunk 26 above)
Comment (andrepestana-aot) [reply sources/packages/backend/apps/db-migrations/src/migrations/1705441135641-RemovePDVerifiedColumn.ts]: I reverted the changes for that file and created a new one with kebab case.

File: sources/packages/backend/apps/db-migrations/src/sql/Student/Add-StudentPDVerified.sql
[the remark this answers is not in the document]
Comment (andrepestana-aot) [reply sources/packages/backend/apps/db-migrations/src/sql/Student/Add-StudentPDVerified.sql:1]: I reverted the changes for that file and created a new one with kebab case.

File: sources/packages/backend/apps/db-migrations/src/sql/Student/Remove-StudentPDVerified.sql
Comment (andrepestana-aot) [reply sources/packages/backend/apps/db-migrations/src/sql/Student/Remove-StudentPDVerified.sql:1]: I reverted the changes for that file and created a new one with kebab case.

File: sources/packages/web/src/views/institution/locations/confirmation-of-enrollment/ApplicationDetailsForCOE.vue [hunk 29]
```diff
@@ -135,6 +136,12 @@ export default defineComponent({
       initialData.value.disbursementDate = dateOnlyLongString(
         initialData.value.disbursementDate,
       );
+      initialData.value.disabilityProfileStatusDescription =
+        disabilityStatusToDisplay(initialData.value.disabilityProfileStatus);
+      initialData.value.disabilityApplicationStatusDescription =
+        ApplicationDisabilityStatusDescription[
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/views/institution/locations/confirmation-of-enrollment/ApplicationDetailsForCOE.vue:142]: Is it not the case for `ApplicationDisabilityStatusDescription` to follow the same composable pattern from `disabilityStatusToDisplay`?

**Review by andrewsignori-aot — COMMENTED** [review]
Great work, only minor comments.

File: sources/packages/web/src/views/institution/locations/confirmation-of-enrollment/ApplicationDetailsForCOE.vue (hunk 29 above)
Comment (andrepestana-aot) [reply sources/packages/web/src/views/institution/locations/confirmation-of-enrollment/ApplicationDetailsForCOE.vue]: ok, changed to composable similar to `disabilityStatusToDisplay`.

File: sources/packages/backend/libs/integrations/src/sfas-integration/sfas-files/sfas-individual-record.ts (hunk 11 above)
Comment (andrepestana-aot) [reply sources/packages/backend/libs/integrations/src/sfas-integration/sfas-files/sfas-individual-record.ts:16]: I see your point and I think it's valid but I won't add any tests for now. We may check if there is already a ticket for that, otherwise I can create a ticket to test it.

File: sources/packages/web/src/composables/useFormatters.ts [hunk 30]
```diff
@@ -363,6 +364,27 @@ export function useFormatters() {
     }
   };
 
+  /**
+   * Convert the application disability status to be displayed
+   * in an equivalent user friendly description.
+   * @param disabilityStatus disability status.
```
Comment (andrewsignori-aot) [inline sources/packages/web/src/composables/useFormatters.ts:370]: The param name must be adjusted from `disabilityStatus` to `applicationDisabilityStatus`.

File: sources/packages/backend/apps/db-migrations/src/migrations/1705441135641-RemovePDVerifiedColumn.ts (hunk 26 above)
Comment (andrewsignori-aot) [reply sources/packages/backend/apps/db-migrations/src/migrations/1705441135641-RemovePDVerifiedColumn.ts]: > Yes, maybe we should be even more generic and use like "change" and "rollback" because sometimes in the db change we do an update or delete.

Do you mean to use  "change" and "rollback" for the `down` script?

**Review by dheepak-aot — APPROVED** [review]
Thanks for making the changes 👍
 Great work👏

**Review by andrewsignori-aot — APPROVED** [review]
Thanks for taking care of the last-minute AC change, great work, looks good 👍

**Review by sh16011993 — APPROVED** [review]
Nice Work @andrepestana-aot 👍
